vslib/src/sai_vs_hostintf.cpp

```diff
@@ -36,10 +37,278 @@ typedef struct _hostif_info_t
+        _In_ sai_object_id_t port_id,
+        _Inout_ sai_object_id_t &bridge_id,
+        _Inout_ sai_object_id_t &bridge_port_id,
+        _Inout_ sai_fdb_entry_bridge_type_t &bridge_type)
+{
+    SWSS_LOG_ENTER();
+
+    bridge_id = SAI_NULL_OBJECT_ID;
+    bridge_port_id = SAI_NULL_OBJECT_ID;
+    bridge_type = SAI_FDB_ENTRY_BRIDGE_TYPE_1Q;
+
+    sai_object_id_t switch_id = sai_switch_id_query(port_id);
+
+    auto &objectHash = g_switch_state_map.at(switch_id)->objectHash.at(SAI_OBJECT_TYPE_BRIDGE_PORT);
+
+    // iterate via all bridge ports to find match on port id
+
+    for (auto it = objectHash.begin(); it != objectHash.end(); ++it)
+    {
+        sai_object_id_t bpid;
+
+        sai_deserialize_object_id(it->first, bpid);
+
+        sai_attribute_t attr;
+
+        attr.id = SAI_BRIDGE_PORT_ATTR_PORT_ID;
+
+        sai_status_t status = vs_generic_get(SAI_OBJECT_TYPE_BRIDGE_PORT, bpid, 1, &attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            continue;
+        }
+
+        if (port_id != attr.value.oid)
+        {
+            // this is not expected port
+            continue;
+        }
+
+        bridge_port_id = bpid;
+
+        SWSS_LOG_DEBUG("found bridge port %s for port %s",
+                sai_serialize_object_id(bridge_port_id).c_str(),
+                sai_serialize_object_id(port_id).c_str());
+
+        attr.id = SAI_BRIDGE_PORT_ATTR_BRIDGE_ID;
+
+        status = vs_generic_get(SAI_OBJECT_TYPE_BRIDGE_PORT, bridge_port_id, 1, &attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            break;
+        }
+
+        bridge_id = attr.value.oid;
+
+        SWSS_LOG_DEBUG("found bridge %s for port %s",
+                sai_serialize_object_id(bridge_id).c_str(),
+                sai_serialize_object_id(port_id).c_str());
+
+        attr.id = SAI_BRIDGE_ATTR_TYPE;
+
+        status = vs_generic_get(SAI_OBJECT_TYPE_BRIDGE, bridge_id, 1, &attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            break;
+        }
+
+        bridge_type = (sai_fdb_entry_bridge_type_t)attr.value.s32;
+
+        SWSS_LOG_DEBUG("bridge %s type is %d",
+                sai_serialize_object_id(bridge_id).c_str(),
+                bridge_type);
+
+        break;
+    }
+}
+
+void process_packet_for_fdb_event(
+        _In_ const uint8_t *buffer,
+        _In_ size_t size,
+        _In_ const std::shared_ptr<hostif_info_t> &info)
+{
+    MUTEX();
+
+    SWSS_LOG_ENTER();
+
+    uint32_t frametime = (uint32_t)time(NULL);
+
+    /*
+     * We add +2 in case if frame contains 1Q VLAN tag.
+     */
+
+    if (size < (sizeof(ethhdr) + 2))
+    {
+        SWSS_LOG_WARN("ethernet frame is too small: %zu", size);
+        return;
+    }
+
+    const ethhdr *eh = (const ethhdr*)buffer;
+
+    uint16_t proto = htons(eh->h_proto);
+
+    uint16_t vlan_id = DEFAULT_VLAN_NUMBER;
+
+    switch (proto)
+    {
+        case ETH_P_IP:
+        case ETH_P_IPV6:
+        case ETH_P_ARP:
+        case ETH_P_RARP:
```

Comment:
we do not need to limit to these ether types, it can be 8021Q or else.

Inside 8021Q, we can later add QinQ packet process, but that is for the future.

Reply:
removed, accepting all protocols